tests/python_package_test/test_engine.py

```diff
@@ -453,3 +453,17 @@ def test_pandas_categorical(self):
         np.testing.assert_almost_equal(pred0, pred2)
         np.testing.assert_almost_equal(pred0, pred3)
         np.testing.assert_almost_equal(pred0, pred4)
+
+    def test_subset_train_val(self):
+        '''
+        Tests that it's fine to construct a single lgb.Dataframe object,
+        takes subsets of it, and uses the subsets for training and validation
+        '''
+        n = 1000
+        X = np.random.normal(size=n * 2).reshape(n, 2)
+        y = np.random.normal(size=n)
+        tmp_dat = lgb.Dataset(X, y)
+        tmp_dat_train = tmp_dat.subset(np.arange(int(n * .8)))
+        tmp_dat_val = tmp_dat.subset(np.arange(int(n * .8), n))
```

Comment:
This use case of using subsets in training is one that I explicitly care about, so I figured it would be worth having a test for this.  Would it be ok if I just did another PR with a test (probably in the basics.py test file) explicitly testing the ref_chain  logic?